pkg/pillar/scripts/generate-self-signed.sh

```diff
@@ -17,26 +17,50 @@ if [ $# != 2 ]; then
 fi
 lifetime=$1
 output_base=$2
+temp_ecc_key=temp.key.pem
 output_key=${output_base}.key.pem
 output_cert=${output_base}.cert.pem
+TPM_ENABLED_FILE="/persist/config/tpm_in_use"
 csr=${output_base}.csr
-dir=$(dirname "$output_key")
+dir=$(dirname "$output_cert")
 if [ ! -d "$dir" ]; then
     echo "Directory does not exist: $dir"
     exit 1
 fi
 
-subject="/C=US/ST=California/L=Santa Clara/O=Zededa, Inc/CN=$(basename "$output_base")"
-openssl ecparam -genkey -name prime256v1 -out "$output_key"
-openssl req -new -sha256 -subj "$subject" -key "$output_key" -out "$csr"
+#The CSR content
+subject="/C=US/ST=California/L=Santa Clara/O=Zededa, Inc/CN=onboard"
+
+if [ -f $TPM_ENABLED_FILE ]; then
```

Comment:
I think it would be better to pass in a new flag argument to the script to have it use the TPM

Reply:
Addressed in the latest commit.